File: src/nl/rubensten/texifyidea/algorithm/Algorithm.java
package nl.rubensten.texifyidea.algorithm;

/**
 * Why did I make an interface for this?
 * <p>
 * Cuz reasons.
 * <p>
 * Information about the algorithm can only be obtained after executing it with
 * {@link Algorithm#execute()}.
 *
 * @author Ruben Schellekens
 */
public interface Algorithm {

    /**
     * Executes the algorithm.
     * <p>
     * Information about the algorithm can only be obtained after executing it.
     */
    void execute();
}

File: src/nl/hannahsten/texifyidea/algorithm/Algorithm.kt
package nl.hannahsten.texifyidea.algorithm

/**
 * Why did I make an interface for this?
 *
 *
 * Cuz reasons.
 *
 *
 * Information about the algorithm can only be obtained after executing it with
 * [Algorithm.execute].
 *
 * @author Hannah Schellekens
 */
interface Algorithm {

    /**
     * Executes the algorithm.
     *
     *
     * Information about the algorithm can only be obtained after executing it.
     */
    fun execute()
}